packages/examples/cvat/exchange-oracle/tests/api/test_exchange_api.py

```diff
@@ -0,0 +1,283 @@
+            patch("src.endpoints.exchange.cvat_api.get_user_id") as mock_get_user,
+        ):
+            mock_get_user.return_value = 1
+            response = client.put(
+                "/register",
+                headers={"signature": "sample"},
+                json={"wallet_address": user_address, "cvat_email": cvat_email_2},
+            )
+
+        user = response.json()
+        assert response.status_code == 200
+        mock_remove_user.assert_called_once()
+        assert user["wallet_address"] == user_address
+        assert user["cvat_email"] == cvat_email_2
+
+        # Existing user
+        cvat_email_2 = "test2@hmt.ai"
+        response = client.put(
+            "/register",
+            headers={"signature": "sample"},
+            json={"wallet_address": user_address, "cvat_email": cvat_email_2},
+        )
+
+        user = response.json()
+        assert response.status_code == 200
+        mock_remove_user.assert_called_once()
+        assert user["wallet_address"] == user_address
+        assert user["cvat_email"] == cvat_email_2
+
+
+def test_register_400(client: TestClient) -> None:
+    with SessionLocal.begin() as session:
+        # User with different wallet address
+        user = User(
+            wallet_address="0x86e83d346041E8806e352681f3F14549C0d2BC61",
+            cvat_email="test@hmt.ai",
+            cvat_id=1,
+        )
+        session.add(user)
+        session.commit()
+        response = client.put(
+            "/register",
+            headers={"signature": "sample"},
+            json={"wallet_address": user_address, "cvat_email": cvat_email},
+        )
+        assert response.status_code == 400
+        assert response.json() == {"message": "User already exists"}
+
+
+def test_register_401(client: TestClient) -> None:
+    response = client.put(
+        "/register",
+        headers={"signature": "test"},
+        json={"wallet_address": user_address, "cvat_email": cvat_email},
+    )
+
+    assert response.status_code == 401
+    assert response.json() == {"message": "Unauthorized"}
+
+    response = client.put(
+        "/register", json={"wallet_address": user_address, "cvat_email": cvat_email}
+    )
+
+    # Send a request without a signature
+    assert response.status_code == 400
+    assert response.json() == {"errors": [{"field": "signature", "message": "field required"}]}
+
+
+def test_create_assignment_200(client: TestClient) -> None:
+    session = SessionLocal()
+    session.begin()
+    cvat_project_1, _, cvat_job_1 = create_project_task_and_job(
+        session, "0x86e83d346041E8806e352681f3F14549C0d2BC67", 1
+    )
+    user = User(
+        wallet_address=user_address,
+        cvat_email="test@hmt.ai",
+        cvat_id=1,
+    )
+    session.add(user)
+    session.commit()
+    with (
+        open("tests/utils/manifest.json") as data,
+        patch("src.services.exchange.get_escrow_manifest") as mock_get_manifest,
+        patch("src.services.exchange.cvat_api"),
```

Comment:
Probably, you should also check that CVAT is called here. A job assignee should be changed after this call.